metis-indexing/src/main/java/eu/europeana/indexing/common/fullbean/FieldInputUtils.java

```diff
@@ -190,6 +213,52 @@ static String getResourceString(ResourceType obj) {
     return array.length > 0 ? array[0] : null;
   }
 
+  /**
+   * Gets literal value string.
+   *
+   * @param obj the LiteralType object
+   * @return a string from the LiteralType object
+   */
+  static String getLiteralValueString(LiteralType obj) {
+    return Optional.ofNullable(obj)
+                   .map(LiteralType::getString)
+                   .orElse(null);
+  }
+
+  /**
+   * Gets resource literal value string.
+   *
+   * @param obj the ResourceOrLiteralType
+   * @return the ResourceLiteralType object
+   */
+  static String getResourceOrLiteralValueString(ResourceOrLiteralType obj) {
+    return Optional.ofNullable(obj)
+                   .map(ResourceOrLiteralType::getResource)
+                   .map(Resource::getResource)
+                   .orElse(Optional.ofNullable(obj)
+                       .map(ResourceOrLiteralType::getString)
+                       .orElse(null));
+  }
+
+  /**
+   * Create string list from type list list.
+   *
+   * @param <T> the type parameter
+   * @param list the list
+   * @param valuesGetter the values getter
+   * @return the list
+   */
+  public static <T> List<String> createStringListFromTypeList(List<T> list,
+      Function<T, String> valuesGetter) {
+
+    return Optional.ofNullable(list)
+                   .orElse(Collections.emptyList())
+                   .stream()
+                   .filter(Objects::nonNull)
+                   .map(valuesGetter)
+                   .toList();
```

Comment:
For consistency, I think this method should be rethought. Perhaps it is easiest to remove it and have callers use the various `...ListToArray` methods instead.

Reply:
the reasoning behind this, is that the type List was not supported before, and that is why didn't opted for reuse. Having context explained over how delicate the methods are with reuse the same validations will propagate everywhere. Done. removed and added the LiteralType method